mitmproxy/addons/proxyserver.py

```diff
@@ -31,9 +31,9 @@
 
 class Servers:
     def __init__(self, manager: ServerManager):
-        self.updating = signals.AsyncSignal(lambda old_modes, new_modes: None)
+        self.updating = signals.AsyncSignal(lambda: None)
         """"Notified before any instances are started or stopped"""
-        self.updated = signals.AsyncSignal(lambda old_modes, new_modes: None)
+        self.updated = signals.AsyncSignal(lambda: None)
```

Comment:
If I may chip in here: The original idea behind the two in #5512 was to support UI changes, like showing a spinner while updating, for `updating` listeners to alter `new_modes` in order to prevent updates, and to improve performance if someone needed a diff of changed servers.
If that all isn't necessary, one `updated` signal without any argument suffices. (Probably should call it `changed` then, to match `options`.)

Reply:
The reason I simplified this was because I modified `new_modes` to be a dict, which then made me trace uses of the signal to adjust those as well. I'm happy to add whatever we need here once there is the first concrete use case in mitmproxy. :)